crates/re_renderer/examples/picking.rs

```diff
@@ -0,0 +1,205 @@
+        let point_count = 10000;
+        let points_positions = (0..point_count)
+            .map(|_| {
+                glam::vec3(
+                    rnd.gen_range(random_point_range.clone()),
+                    rnd.gen_range(random_point_range.clone()),
+                    rnd.gen_range(random_point_range.clone()),
+                )
+            })
+            .collect_vec();
+        let points_radii = std::iter::repeat(Size::new_scene(0.08))
+            .take(point_count)
+            .collect_vec();
+        let points_colors = (0..point_count)
+            .map(|_| random_color(&mut rnd))
+            .collect_vec();
+        let points_picking_ids = (0..point_count as u64)
+            .map(PickingLayerInstanceId)
+            .collect_vec();
+
+        Picking {
+            points_positions,
+            points_radii,
+            points_colors,
+            points_picking_ids,
+            scheduled_picking_rects: Vec::new(),
+            picking_position: glam::UVec2::ZERO,
+        }
+    }
+
+    fn draw(
+        &mut self,
+        re_ctx: &mut re_renderer::RenderContext,
+        resolution: [u32; 2],
+        time: &framework::Time,
+        pixels_from_point: f32,
+    ) -> Vec<framework::ViewDrawResult> {
+        self.handle_incoming_picking_data(re_ctx, time.seconds_since_startup());
+
+        let mut view_builder = ViewBuilder::default();
+
+        // TODO(#1426): unify camera logic between examples.
+        let camera_position = glam::vec3(1.0, 3.5, 7.0);
+
+        view_builder
+            .setup_view(
+                re_ctx,
+                TargetConfiguration {
+                    name: "OutlinesDemo".into(),
+                    resolution_in_pixel: resolution,
+                    view_from_world: macaw::IsoTransform::look_at_rh(
+                        camera_position,
+                        glam::Vec3::ZERO,
+                        glam::Vec3::Y,
+                    )
+                    .unwrap(),
+                    projection_from_view: Projection::Perspective {
+                        vertical_fov: 70.0 * std::f32::consts::TAU / 360.0,
+                        near_plane_distance: 0.01,
+                    },
+                    pixels_from_point,
+                    outline_config: None,
+                    ..Default::default()
+                },
+            )
+            .unwrap();
+
+        let picking_rect_size = 32;
+        self.scheduled_picking_rects.push(
+            view_builder
+                .schedule_picking_readback(
+                    re_ctx,
+                    self.picking_position.as_ivec2()
+                        - glam::ivec2(picking_rect_size / 2, picking_rect_size / 2),
+                    picking_rect_size as u32,
+                    false,
+                )
+                .unwrap(),
+        );
+
+        let mut builder = PointCloudBuilder::<()>::new(re_ctx);
+
+        // Split into two batches to test picking of multiple batches.
```

Comment:
NOW that makes sense. It would have made sense earlier if `struct Picking` contained two batches of points instead of just one batch